java/BUILD.bazel

```diff
@@ -0,0 +1,242 @@
+load("//java/tools/checkstyle:checkstyle.bzl", "checkstyle_test")
+load("//bazel:ray.bzl", "flatbuffer_java_library")
+
+java_binary(
+    name = "bazel_deps",
+    main_class = "com.github.johnynek.bazel_deps.ParseProject",
+    runtime_deps = ["@bazel_deps//jar"],
+)
+
+java_import(
+    name = "org_ray_ray_java_root",
+    jars = [
+        "liborg_ray_ray_api.jar",
+        "liborg_ray_ray_api-src.jar",
+        "liborg_ray_ray_runtime.jar",
+        "liborg_ray_ray_runtime-src.jar",
+        "liborg_ray_ray_runtime_test.jar",
+        "liborg_ray_ray_runtime_test-src.jar",
+        "liborg_ray_ray_test.jar",
+        "liborg_ray_ray_test-src.jar",
+        "org_ray_ray_tutorial_deploy.jar",
+        "org_ray_ray_tutorial_deploy-src.jar",
+        "AllTests_deploy.jar",
+        "AllTests_deploy-src.jar",
+    ],
+    deps = [
+        ":org_ray_ray_api",
+        ":org_ray_ray_runtime",
+        ":org_ray_ray_tutorial",
+        ":org_ray_ray_test",
+        ":org_ray_ray_runtime_test",
+    ],
+)
+
+java_library(
+    name = "org_ray_ray_api",
+    srcs = glob(["api/src/main/java/**/*.java"]),
+    deps = [
+        "@mvnorg_slf4j_slf4j_log4j12//jar",
+        "@mvnlog4j_log4j//jar",
+        "@mvnjavax_xml_bind_jaxb_api//jar",
+        "@mvncom_sun_xml_bind_jaxb_core//jar",
+        "@mvncom_sun_xml_bind_jaxb_impl//jar",
+    ],
+)
+
+checkstyle_test(
+    name = "org_ray_ray_api-checkstyle",
+    target = ":org_ray_ray_api",
+    size = "small",
+)
+
+java_library(
+    name = "org_ray_ray_runtime",
+    srcs = glob([
+        "runtime/src/main/java/**/*.java"
+    ]) + [
+        ":generate_java_gcs_fbs",
+    ],
+    resources = glob(["runtime/src/main/resources/**"]),
+    visibility = ["//visibility:public"],
+    data = ["//:raylet_library_java"],
+    deps = [
+        ":org_ray_ray_api",
+        "@plasma//:org_apache_arrow_arrow_plasma",
+        "@mvnorg_slf4j_slf4j_api//jar",
+        #"@mvncom_google_code_gson_gson//jar",
+        "@mvncom_fasterxml_jackson_core_jackson_core//jar",
+        "@mvnorg_apache_commons_commons_pool2//jar",
+        "@mvnorg_javassist_javassist//jar",
+        "@mvnorg_objenesis_objenesis//jar",
+        "@mvncom_typesafe_config//jar",
+        "@mvnorg_apache_commons_commons_lang3//jar",
+        "@mvnde_ruedigermoeller_fst//jar",
+        "@mvncom_github_davidmoten_flatbuffers_java//jar",
+        "@mvnredis_clients_jedis//jar",
+        "@mvncommons_io_commons_io//jar",
+        "@mvncom_google_guava_guava//jar",
+        "@mvnnet_lingala_zip4j_zip4j//jar",
+        "@mvnorg_slf4j_slf4j_log4j12//jar",
+        "@mvnorg_ini4j_ini4j//jar",
+        "@mvnorg_ow2_asm_asm//jar",
+    ],
+)
+
+checkstyle_test(
+    name = "org_ray_ray_runtime-checkstyle",
```

Comment:
Shall we change `-` to `_` for consistent naming convention?

Reply:
xxx-checkstyle is the naming format required by the checkstyle tool